plugins/UFM_NDT_Plugin/ufm_sim_web_service/merger_resources.py

```diff
@@ -198,52 +209,69 @@ def run_ibdiagnet_ndt_compare(self, ndt_file_name):
             if error_message:
+                report_content["error"] = error_message
+                report_content["status"] = "Verification failed."
                 raise ValueError(error_message)
             ndt_links = set()
             ndt_links_reversed = set()
             error_message = parse_ndt_file(ndt_links, ndt_file_name,
                             self.switch_patterns + self.host_patterns,
                             ndt_links_reversed, True)
             if error_message:
+                report_content["error"] = error_message
+                report_content["status"] = "Verification failed."
                 raise ValueError(error_message)
             # compare NDT with ibdiagnet
             # create report
-            if not links_info or not create_topoconfig_file(links_info, ndt_file_name,
-                            self.switch_patterns + self.host_patterns):
-                #this is the structure that contains names of the nodes and ports and GUIDs
-                # on base of this struct should be created topconfig file
-                # in case of failure - non eed to continue - crewate error report and return
-                report_content["error"] = "Failed to create topoconfig file",
-                report_content["status"] = "Failed on topoconfig creation",
-                report_content["timestamp"] = self.timestamp,
-                report_content["report"] = {}
-                report_content["NDT_file"] = os.path.basename(ndt_file_name)
+            if not links_info:
+                report_content["error"] = "Failed to create topoconfig file. No links found"
+                report_content["status"] = "Verification failed."
                 raise ValueError(report_content["error"])
             report_content = compare_topologies_ndt_ibdiagnet(self.timestamp,
                                                               ibdiagnet_links,
                                                               ibdiagnet_links_reverse,
                                                               ndt_links,
                                                               ndt_links_reversed)
             report_content["NDT_file"] = os.path.basename(ndt_file_name)
+            topoconfig_creation_status, message, failed_ports = create_topoconfig_file(links_info,
+                      ndt_file_name, self.switch_patterns + self.host_patterns)
+            # If open comments below - remove / comment next if statement
+            if not topoconfig_creation_status or failed_ports:
+                report_content["error"] = message
+                report_content["status"] = "Verification failed."
+                logging.error(message)
+                raise ValueError(report_content["error"])
+            # TODO: ATB additional section/category in report for labels that
+            # failed to be converted to port number. Uncomment if will be request for such info
```

Comment:
according to our discussion, the new category is not required, please cleanup the commented out code...